app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb

```diff
@@ -80,7 +76,7 @@ def find_content_view_puppet_module
     end
 
     def find_puppet_module
-      @puppet_module = PuppetModule.find(params[:id]) if params[:id]
+      @puppet_module = PuppetModule.find_by(puppet_module_params)
```

Comment:
I think you can drop this `find_puppet_module` method since create is redefining `@puppet_module`.